packages/ddp-client/common/livedata_connection.js

```diff
@@ -1721,7 +1721,7 @@ export class Connection {
 
     self.onReconnect && self.onReconnect();
     DDP._reconnectHook.each(callback => {
-      callback(self);
+      callback(self, !self.shouldRetryMethods && self.errorNoRetry);
```

Comment:
If we hope to ever upstream this, we should probably store and forward all disconnect errors, not just these special ones. Especially if you say in documentation comment `the last error (if there is one) that caused the connection to break.` - that implies all kinds of possible errors.

Reply:
@OleksandrChekhovskyi I changed the documentation to specify that only errors that caused the `shouldRetryMethods` to true will be forwarded. I don't think we should forward all errors, since the cases we added are the only "exceptional" ones AFAIK

Reply:
The idea is that this is a general-purpose hook, so it should report all kinds of errors if it does at all. But maybe if we want a very specific thing, it would be just easier to set up a separate hook? (for our specific Favro use, not intended for upstreaming).

Reply:
fixed - now we are passing all errors (if any) to the hooks, this includes the old `noRetryError` ones and any other error that could make the server send the `close` request to the client.